src/ring_theory/power_basis.lean

```diff
@@ -203,22 +195,24 @@ begin
 end
 
 @[simp]
-lemma nat_degree_minpoly_gen (pb : power_basis A S) :
+lemma nat_degree_minpoly_gen [nontrivial A] (pb : power_basis A S) :
   nat_degree (minpoly_gen pb) = pb.dim :=
 nat_degree_eq_of_degree_eq_some pb.degree_minpoly_gen
 
-lemma minpoly_gen_monic (pb : power_basis A S) : monic (minpoly_gen pb) :=
+@[simp]
+lemma minpoly_gen_eq (pb : power_basis A S) : pb.minpoly_gen = minpoly A pb.gen :=
```

Comment:
@Vierkantor Since you original wrote `minpoly_gen`: now that I showed it's always equal to `minpoly`, do you think it's a good idea to replace all uses of `pb.minpoly_gen` by `minpoly A pb.gen`? The `simp_nf` linter is giving me trouble because now the former always simplifies to the latter. Maybe it shouldn't be a simp lemma? Because the RHS mentions `A` explicitly while the LHS doesn't need to. If we don't keep `minpoly_gen`, we can still keep a lemma stating that `minpoly A pb.gen` is equal to the current definition of `pb.minpoly_gen` (`= X ^ pb.dim - ...`).

Reply:
I guess I'll take the minimally disruptive option and simply remove a bunch of `simp` attributes ...

Reply:
I feel like removing `minpoly_gen` seems reasonable, but I guess that's probably extra work...

Reply:
I think `minpoly_gen_eq` is a good simp lemma, since we use `minpoly` a lot throughout the library. Since there is already a corresponding lemma for `minpoly A pb.gen` it should be safe to remove the `@[simp]` attributes where the linter complains, except we need a new corresponding to `degree_minpoly_gen`.

Reply:
Also I think completely removing `minpoly_gen` might make it hard to prove that `pb.gen` is indeed integral, which is still needed to prove this result.